cafebabel/static/js/edit-article.js

```diff
@@ -27,68 +39,108 @@ class Tags {
   _createTag(tagValue, index) {
     const li = document.createElement('li')
     const input = document.createElement('input')
+    const a = document.createElement('a')
     input.type = 'hidden'
     input.name = `tag-${index + 1}`
     input.setAttribute('list', 'tags')
     input.value = tagValue
     li.append(tagValue)
     li.append(input)
+    li.append(a)
 
     return li
   }
 }
 
-const tagButtonAdd = document.querySelector('.tags button.add')
-
-tagButtonAdd.addEventListener('click', event => {
-  event.preventDefault()
-  const submission = event.target.previousSibling.value
-  const tags = new Tags(document.querySelector('.tags'))
-  if (tags.checkSubmission(submission)) tags.add(submission)
-  displayTags(tags)
-  event.target.previousSibling.value = ''
-})
-
-function handleSuggestion(tag) {
-  const tagsSuggestion = document.querySelector('#tags-suggestions')
-  tagsSuggestion.innerHTML = ''
-  const option = document.createElement('option')
-  option.value = tag.name
-  tagsSuggestion.appendChild(option)
+class Effect {
+  static fadeIn(element) {
+    element.style.opacity = 0
+    ;(function fade() {
+      let val = parseFloat(element.style.opacity)
+      if (!((val += 0.03) > 1)) {
+        element.style.opacity = val
+        requestAnimationFrame(fade)
+      }
+    })()
+  }
+  static fadeOut(element) {
+    element.style.opacity = 1
+    return new Promise((resolve, reject) => {
+      ;(function fade() {
+        if ((element.style.opacity -= 0.03) < 0) {
+          element.style.display = 'none'
+          resolve()
+        } else {
+          requestAnimationFrame(fade)
+        }
+      })()
+    })
+  }
 }
 
-document
-  .querySelector('.tags input[name=tag-new]')
-  .addEventListener('keyup', event => {
-    /* Return if arrow up, arrow down or enter are pressed */
-    if (event.keyCode == 40 || event.keyCode == 38 || event.keyCode == 13)
-      return
-    document.querySelector('#tags-suggestions').innerHTML = ''
-    const submission = event.target.value
-    const languages = document.querySelector('#language')
-    const language = languages.options[languages.selectedIndex].value
-    if (submission.length < 3) return
-    request(`/article/tag/suggest/?language=${language}&terms=${submission}`)
-      .then(response => response.forEach(handleSuggestion))
-      .catch(console.error.bind(console))
-  })
+class Listener {
```

Comment:
This seems very specific to tags while the class sounds very generic.
Why not going for a name like `TagEventListener`?